packages/yew/src/functional/hooks/use_async.rs

```diff
@@ -0,0 +1,56 @@
+use std::{fmt, future::Future};
+
+use crate::{functional::hook, use_effect_with, use_state, UseStateHandle};
+
+#[hook]
+pub fn use_async<Arg, F, Fut, Res>(deps: Arg, f: F) -> UseAsyncHandle<Res>
+where
+    Arg: 'static + PartialEq,
+    F: 'static + FnOnce(&Arg) -> Fut,
+    Fut: 'static + Future<Output = Res>,
+    Res: 'static,
+{
+    let result = use_state(|| None);
+    use_effect_with(deps, {
```

Comment:
Hmm so TBH I have very little knowledge of how SSR is supposed to work, and zero experience with it — I'm using yew to develop cross-platform pseudo-heavy applications rather than lightweight webapps.

Do you see `use_async` being useful with SSR? And if yes, is there any other hook I could use to define dependencies?

Reply:
I was thinking that perhaps you want to asynchronously fetch data on the server to render. But on second thought, this was a bit hasty because the component will be only rendered once in SSR anyway, so any resolved result will likely never make it into the component anyway if this works just similar to `use_memo` and just spawns the future, but then continues to render. The solution could be to just not poll the future on the server at all.

Note, because you also mention it in the other comment about `use_future`: Suspensions do work on SSR (see there for a small explanation on what suspensions are) and wait for the future to complete, i.e. the suspension to resolve before rendering the component into the stream.